src/vdom/diff.js

```diff
@@ -124,6 +124,8 @@ function idiff(dom, vnode, context, mountAll, componentRoot) {
 		props = out[ATTR_KEY] || (out[ATTR_KEY] = {}),
 		vchildren = vnode.children;
 
+	for (let a=out.attributes, i=a.length; i--; ) props[a[i].name] = a[i].value;
```

Comment:
we'll need to wrap this in a check for props. If the propcache already exists on an element, skipping attribute checks is a fairly major performance gain.